bin/sozo/src/commands/events.rs

```diff
@@ -42,16 +42,29 @@ pub struct EventsArgs {
 
 impl EventsArgs {
     pub fn run(self, config: &Config) -> Result<()> {
+        let env_metadata = utils::load_metadata_from_config(config)?;
         let ws = scarb::ops::read_workspace(config.manifest_path(), config)?;
-
-        let env_metadata = if config.manifest_path().exists() {
-            dojo_metadata_from_workspace(&ws).and_then(|inner| inner.env().cloned())
-        } else {
-            None
-        };
-
         let manifest_dir = ws.manifest_path().parent().unwrap().to_path_buf();
+        let provider = self.starknet.provider(env_metadata.as_ref())?;
+
+        let event_filter = events::get_event_filter(
+            self.from_block,
+            self.to_block,
+            self.events,
+            self.world.world_address,
+        );
 
-        config.tokio_handle().block_on(events::execute(self, env_metadata, &manifest_dir))
+        let _ = config.tokio_handle().block_on(async {
+            events::execute(
```

Comment:
```suggestion
            events::parse(
```
Execute is too generic, and we may add some functions into `ops/events` in the future.